### PR TITLE
Unit-test with spire integration

### PR DESCRIPTION
**Pull request by pxp928**
Unit tests added for the below files:
- [u] pkg/apis/config/feature_flags.go
- [u] pkg/apis/pipeline/v1beta1/taskrun_types.go
- [u] pkg/pod/pod.go

Added helper function - `IsTaskRunResultVerified`
Cleaned up if statemement in function `filterResultsAndResources`

### CODE REVIEW

File: pkg/pod/pod_test.go [hunk 1]
```diff
@@ -1788,6 +1788,177 @@ func TestPodBuildwithAlphaAPIEnabled(t *testing.T) {
...
+				VolumeSource: corev1.VolumeSource{
+					CSI: &corev1.CSIVolumeSource{
+						Driver: "csi.spiffe.io",
+					},
+				},
+			}),
+			ActiveDeadlineSeconds: &defaultActiveDeadlineSeconds,
+		},
+		wantAnnotations: map[string]string{
+			SpiffeIdAnnotation: fmt.Sprintf("ns/%v/taskrun/%v", "default", "taskrun-name"),
```
Comment (lumjjb) [inline pkg/pod/pod_test.go:1874]: hmm, i forget where we add that annotation, is this annotation key defined anywhere?

**Review by lumjjb — APPROVED** [review]
lgtm, minor nits! nice tests!

File: pkg/apis/config/testdata/feature-flags-all-flags-set.yaml [hunk 2]
```diff
@@ -25,3 +25,5 @@ data:
   enable-custom-tasks: "true"
   scope-when-expressions-to-task: "true"
   enable-api-fields: "alpha"
+  enable-spire: "true"
+
```
[the remark this answers is not in the document]
Comment (pxp928) [reply pkg/apis/config/testdata/feature-flags-all-flags-set.yaml:29]: fixed.

File: pkg/pod/pod_test.go (hunk 1 above)
Comment (pxp928) [reply pkg/pod/pod_test.go:1874]: removed.

File: pkg/pod/status.go [hunk 3]
```diff
@@ -249,15 +249,8 @@ func filterResultsAndResources(results []v1beta1.PipelineResourceResult, spireEn
...
-				if strings.HasSuffix(r.Key, spire.KeySignatureSuffix) {
-					filteredResults = append(filteredResults, r)
-					continue
-				}
-				if r.Key == spire.KeySVID {
-					filteredResults = append(filteredResults, r)
-					continue
-				}
-				if r.Key == spire.KeyResultManifest {
+				switch {
```
[the remark this answers is not in the document]
Comment (pxp928) [reply pkg/pod/status.go:252]: if statement added.

File: pkg/apis/pipeline/v1beta1/taskrun_types_test.go [hunk 4]
```diff
@@ -135,6 +135,27 @@ func TestTaskRunIsCancelled(t *testing.T) {
 	}
 }
 
+func TestTaskRunIsTaskRunResultVerified(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (pxp928) [reply pkg/apis/pipeline/v1beta1/taskrun_types_test.go:138]: add empty condition check.

**Pull request merged**